app/src/androidTest/java/com/github/steroidteam/todolist/AudioRecorderFragmentTest.java

```diff
@@ -0,0 +1,71 @@
+package com.github.steroidteam.todolist;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+import androidx.fragment.app.testing.FragmentScenario;
+import com.github.steroidteam.todolist.view.AudioRecorderFragment;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AudioRecorderFragmentTest {
+
+    @Before
+    public void init() {
+        FragmentScenario<AudioRecorderFragment> scenario =
+                FragmentScenario.launchInContainer(
+                        AudioRecorderFragment.class, null, R.style.Theme_Asteroid);
+    }
+
+    @Test
+    public void toastAndTextWhenRecordingShowsCorrectly() {
+        onView(withId(R.id.record_button)).perform(click());
+        onView(withText(R.string.is_recording))
+                .inRoot(new ToastMatcher())
+                .check(matches(isDisplayed()));
+        onView(withId(R.id.record_text)).check(matches(withText(R.string.stop_record_button)));
+        // Wait to avoid error with Toast messages that is still displayed
+        waitFor(3000);
```

Comment:
Is this wait necessary if you do nothing after in the test ?

Reply:
I put this to avoid that the toast is still displayed for the next test so I think this is useful